internal/lightcone/destruction/shatteredhome/shatteredhome.go

```diff
@@ -0,0 +1,54 @@
+package shatteredhome
+
+import (
+	"github.com/simimpact/srsim/pkg/engine"
+	"github.com/simimpact/srsim/pkg/engine/equip/lightcone"
+	"github.com/simimpact/srsim/pkg/engine/event"
+	"github.com/simimpact/srsim/pkg/engine/info"
+	"github.com/simimpact/srsim/pkg/engine/modifier"
+	"github.com/simimpact/srsim/pkg/engine/prop"
+	"github.com/simimpact/srsim/pkg/key"
+	"github.com/simimpact/srsim/pkg/model"
+)
+
+const (
+	ShatteredHomeCheck key.Modifier = "shattered-home-check"
+	ShatteredHomeBuff  key.Modifier = "shattered-home-buff"
+)
+
+func init() {
+	lightcone.Register(key.ShatteredHome, lightcone.Config{
+		CreatePassive: Create,
+		Rarity:        3,
+		Path:          model.Path_DESTRUCTION,
+		Promotions:    promotions,
+	})
+
+	modifier.Register(ShatteredHomeCheck, modifier.Config{
+		Listeners: modifier.Listeners{
+			OnBeforeHitAll: onBeforeHitAll,
+		},
+	})
+}
+
+// The wear deals 20%/25%/30%/35%/40% more DMG to enemy targets whose HP percentage is greater than 50%.
+func Create(engine engine.Engine, owner key.TargetID, lc info.LightCone) {
+	engine.AddModifier(owner, info.Modifier{
+		Name:   ShatteredHomeCheck,
+		Source: owner,
+		State:  0.15 + 0.05*float64(lc.Imposition),
+	})
+
+}
+
+func onBeforeHitAll(mod *modifier.ModifierInstance, e event.HitStartEvent) {
+	if e.Hit.Defender.CurrentHPRatio() > 0.5 {
+		mod.Engine().AddModifier(mod.Owner(), info.Modifier{
+			Name:   ShatteredHomeBuff,
+			Source: mod.Owner(),
+			Stats:  info.PropMap{prop.AllDamagePercent: mod.State().(float64)},
+		})
```

Comment:
You don't want a new buff modifier here. When inside a `BeforeHit` listener, the stats of the attacker and defender has been snapshotted. Any modifiers applied here wont take effect til the next hit.

Inside `BeforeHit` listeners, you instead want to modify the snapshots directly. In dm they modify `Attacker_AllDamageTypeAddedRatio`, so you'd want to modify the attacker's snapshot:
```go
e.Hit.Attacker.AddProperty(prop.AllDamagePercent, mod.State().(float64))
```

This is similar to the modifiers that happen on In The Night LC: https://github.com/simimpact/srsim/blob/3f1f0c81efa576d3834994ca8307c08a7e25a5c2/internal/lightcone/hunt/inthenight/inthenight.go#L70-L74

Reply:
edited accordingly @unleashurgeek